src/api/common.ts

```diff
@@ -217,3 +218,28 @@ export function validate2xx(status: number): boolean {
 export function validate404(status: number): boolean {
     return validate2xx(status) || status === 404;
 }
+
+export function parseTrackerResponse<T, M extends D2ModelSchemaBase, Fields>(
+    response: HttpClientResponse<T>,
+    trackerKey: TrackedKeys
+) {
+    const { data } = response as HttpClientResponse<
+        T & {
+            instances?: SelectedPick<M, Fields>[];
+            trackedEntities?: SelectedPick<M, Fields>[];
+            events?: SelectedPick<M, Fields>[];
+            enrollments?: SelectedPick<M, Fields>[];
+        }
+    >;
+
+    const responseData = _(data)
+        .omit([trackerKey])
+        .value();
+
+    return {
+        ...responseData,
+        instances: data.instances || data[trackerKey],
+    };
+}
+
+type TrackedKeys = "trackedEntities" | "enrollments" | "events";
```

Comment:
It should be singular. It seems plural, because of the enumeration, but it's describing a value of some type, not some values.